BinaryTree.cpp

```diff
@@ -0,0 +1,70 @@
+#include "BinaryTree.h"
+
+BinaryTree::BinaryTree(int value) : _rootValue(value) {
+    _rootNode->setValue(value);
+}
+
+BinaryTree::~BinaryTree() {
+    delete _rootNode;
+}
+
+void BinaryTree::Insert(int value) {
+    if (value < _rootValue) {
+        if (_leftChild == nullptr) {
+            _leftChild = new TreeNode(value);
+            _rootNode->setLeftChild(_leftChild);
+        } else
+            Insert(value, _leftChild);
+    } else if (value > _rootValue) {
+        if (_rightChild == nullptr) {
+            _rightChild = new TreeNode(value);
+            _rootNode->setRightChild(_rightChild);
+        } else
+            Insert(value, _rightChild);
+    }
+}
+
+TreeNode *BinaryTree::Search(int value) {
```

Comment:
```suggestion
TreeNode* BinaryTree::Search(int value) {
```
Так правильнее. * относится к TreeNode